komodo/yaml_file_type.py

```diff
@@ -38,16 +134,506 @@ class ManifestFile(YamlFile):
     Return the data from 'manifest' YAML, but validate it first.
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.yml = None
+
     def __call__(self, value: str) -> Dict[str, Dict[str, str]]:
-        yaml = super().__call__(value)
+        yml = super().__call__(value)
+        self.validate_manifest_file(yml)
+        return yml
+
+    @staticmethod
+    def validate_manifest_file(manifest_file_content: dict):
         message = (
             "The file you provided does not appear to be a manifest file "
             "produced by komodo. It may be a release file. Manifest files "
             "have a format like the following:\n\n"
             "python:\n  maintainer: foo@example.com\n  version: 3-builtin\n"
             "treelib:\n  maintainer: foo@example.com\n  version: 1.6.1\n"
         )
-        for _, metadata in yaml.items():
-            assert isinstance(metadata, dict), message
-            assert isinstance(metadata["version"], str), message
-        return yaml
+        assert isinstance(manifest_file_content, dict), message
+        errors = []
+        for package_name, metadata in manifest_file_content.items():
+            if not isinstance(metadata, dict):
+                errors.append(f"Invalid metadata for package '{package_name}'")
+                continue
+            if not isinstance(metadata["version"], str):
+                errors.append(
+                    f"Invalid version type in metadata for package '{package_name}'"
+                )
+        handle_validation_errors(errors, message)
+
+
+class RepositoryFile(YamlFile):
+    """
+    Return the data from 'repository' YAML, but validate it first.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.yml = None
+
+    def __call__(self, value: str):
+        self.yml = super().__call__(value)
+        self.validate_repository_file()
+        return self
+
+    def from_yaml_string(self, value: bytes):
+        yaml = YAML()
+        yml = yaml.load(value)
+        self.yml = yml
```

Comment:
suggest `self.yml = YAML().load(value)` to avoid unneeded and awkward-to-name temporary variables.

Reply:
Good idea. I did it in separate lines because I was not sure if we had to set options for the yaml loader before using the load method, but it worked with default options.